### PR TITLE
Update PULL_REQUEST_TEMPLATE.md

### PR DESCRIPTION
**Pull request by oliver-sanders**
Update checklist (will duplicate to 8.0.x if/when approved).

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,27 +1,17 @@
-<!-- Complete this Pull Request template. -->
+<!--
+Thanks for your contribution:
+* Please list any related issues with a "closes" or "addresses" tag.
+* For bugfixes we have maintenance branches e.g. `8.0.x`, please raise separate
+  pull requests against master and the maintenance release branches as appropriate.
+-->
```
Comment (MetRonnie) [inline .github/PULL_REQUEST_TEMPLATE.md:6]: Worth adding "please provide a description"? All too often we're having PRs that just say "closes X" and it takes a fair bit of reading of the original issue + comments to figure out what the change is supposed to do.

**Review by MetRonnie — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (hjoliver) [reply .github/PULL_REQUEST_TEMPLATE.md:6]: Agreed!

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,27 +1,17 @@
...
-- [ ] Already covered by existing tests.
-- [ ] Does not need tests (why?).
-<!-- choose one: -->
-- [ ] Appropriate change log entry included.
-- [ ] No change log entry required (why? e.g. invisible to users).
-<!-- choose one: -->
-- [ ] (master branch) I have opened a documentation PR at cylc/cylc-doc/pull/XXXX.
-- [ ] (7.8.x branch) I have updated the documentation in this PR branch.
-- [ ] No documentation update required.
+- [ ] Tests are included if required.
```
Comment (hjoliver) [inline .github/PULL_REQUEST_TEMPLATE.md:14]:
```suggestion
- [ ] Tests are included (if not, please explain why they are not needed).
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,27 +1,17 @@
...
+- [ ] Changelog is included if required.
```
Comment (hjoliver) [inline .github/PULL_REQUEST_TEMPLATE.md:15]: Explain when they are required?
```suggestion
- [ ] Changelog is included if this is a significant user-facing change
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -1,27 +1,17 @@
...
+- [ ] [Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened if required at cylc/cylc-doc/pull/XXXX.
+- [ ] [if bugfix] PRs raised to both master and the relevant bugfix branch.
```
Comment (hjoliver) [inline .github/PULL_REQUEST_TEMPLATE.md:17]:
```suggestion
- [ ] If this is a bug fix, PRs raised to both master and the relevant maintenance branch.
```

**Review by hjoliver — CHANGES_REQUESTED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -1,27 +1,17 @@
-<!-- Complete this Pull Request template. -->
+<!--
+Thanks for your contribution:
+* Please list any related issues with a "closes" or "addresses" tag.
```
Comment (oliver-sanders) [inline .github/PULL_REQUEST_TEMPLATE.md:3]:
```suggestion
* Please list any related issues with a "closes" or "addresses" tag.
* Please add a helpful description.
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -1,27 +1,18 @@
...
-- [ ] Does not need tests (why?).
-<!-- choose one: -->
-- [ ] Appropriate change log entry included.
-- [ ] No change log entry required (why? e.g. invisible to users).
-<!-- choose one: -->
-- [ ] (master branch) I have opened a documentation PR at cylc/cylc-doc/pull/XXXX.
-- [ ] (7.8.x branch) I have updated the documentation in this PR branch.
-- [ ] No documentation update required.
+- [ ] Tests are included (or explain why tests are not needed).
+- [ ] `CHANGES.md` entry included if this is a significant change
```
Comment (MetRonnie) [inline .github/PULL_REQUEST_TEMPLATE.md:16]:
```suggestion
- [ ] `CHANGES.md` entry included if this is a change that can affect users
```

**Review by MetRonnie — APPROVED** [review]

**Review by hjoliver — APPROVED** [review]

**Pull request merged**